plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2933,6 +2975,14 @@ bool producer_plugin_impl::push_read_only_transaction(
       auto remaining_time_in_read_window_us = _ro_read_window_time_us - time_used_in_read_window_us;
       // Ensure the trx to finish by the end of read-window.
       auto window_deadline = std::min( start + remaining_time_in_read_window_us, block_deadline );
+
+      auto track_exec_time = fc::make_scoped_exit([this,start](){
+         if ( _log.is_enabled( fc::log_level::debug ) ) {
+            // add_time is expensive as it is protected by a mutex.
+            // use it only when log is enabled
+            _ro_all_threads_exec_time.add_time( fc::time_point::now() - start );
+         }
+      });
```

Comment:
Seems like overkill. Would rather just see the add after the `chain.push_transaction` line.

Reply:
We don't count if any exception happens?

Reply:
Should not be possible for that to throw outside the bad_alloc cases which we will exit on anyway.